api/app/helpers/spree/api/api_helpers.rb

```diff
@@ -121,11 +121,18 @@ def required_fields_for(model)
         :id, :number, :state, :order_id, :memo, :created_at, :updated_at
       ]
 
-      @@address_attributes = [
-        :id, :firstname, :lastname, :full_name, :address1, :address2, :city,
-        :zipcode, :phone, :company, :alternative_phone, :country_id, :country_iso,
-        :state_id, :state_name, :state_text
-      ]
+      @@address_base_attributes = [
+        :id, :name, :address1, :address2, :city, :zipcode, :phone, :company,
+        :alternative_phone, :country_id, :country_iso, :state_id, :state_name,
+        :state_text
+      ]
+
+      @@address_attributes = if Spree::Config.use_combined_first_and_last_name_in_address
+                               @@address_base_attributes
+                             else
+                               @@address_base_attributes +
+                                 Spree::Address::DEPRECATED_ATTRS.map(&:to_sym)
```

Comment:
Technically, these are not yet deprecated. Also, we may have other deprecations in the future that will be included/excluded here as well. What about calling this constant in a more semantic way like `SPLIT_NAME_ATTRS` instead?

Reply:
Yes, right now that would be better. I was already thinking about the incoming deprecations but it's premature

Reply:
That sounds perfect!